ExtraDry/ExtraDry.Blazor/Components/Dry/DryTable.razor.cs

```diff
@@ -229,7 +229,7 @@ private void Toggle(ListItemInfo<TItem> item)
 
     private ItemCollection<TItem> InternalItems { get; } = [];
 
-    private IEnumerable<ListItemInfo<TItem>> ShownItems => InternalItems.Where(e => e.IsShown);
+    public IEnumerable<ListItemInfo<TItem>> ShownItems => InternalItems.Where(e => e.IsShown);
```

Comment:
Pretty nervous about making this `public`.  The class `ListItemInfo` is intended to be an internal class only and is reflected by being in a namespace named 'internal'.  However, it is technically 'public' as it is used for Razor properties.  I couldn't figure out a way to make razor components completely internal so settled on the namespace identifier.  I see this pattern leaking out of some Microsoft libraries as well.   

In the sample App, I can only see it being used to access the actual Item.  For exposing, it might be better to have `IEnumerable<TItem> VisibleItems => InternalItems.Where(e => e.IsShown).Select(e => e.Item);`?   

Then, looking deeper, I don't think it was ever designed for this scenario.  It has a concept of "grouped" item, which was the first implementation of hierarchy, which might not even work any more.  So, this concept might be obsolete and should be removed.  

What about `DryTable.RefreshItem(TItem itemToRefreshIfInTable)`?

Reply:
I did try something like `IEnumerable<TItem> VisibleItems => InternalItems.Where(e => e.IsShown).Select(e => e.Item);`, but when the item was replaced in the `CompanyList` page i.e. `existingCompany = updatedCompany` the changes weren't reflected in the list, even after a `StateHasChanged()` call.

I did consider an update method similar to your suggestion but was unsure how to retrieve the item to be refreshed.  Something like `var itemInfo = InternalItems.FirstOrDefault(itemInfo => itemInfo.Item?.Equals(updatedItem) ?? false)` would require the `TItem` class to override `Equals`.  This didn't feel like the right approach.

I wasn't keen on making this `public` but it seemed like the right approach.

Reply:
Based on our conversation I've set this back to `private` and added a refresh method to the `DryTable`.